logos/tests/unit/capacity/test_best_first_scenarios.py

```
@@ -1177,3 +1177,128 @@ def _demand_for(model):
+            provider_id=1,
+            required_gpus=frozenset({0}),
+            per_gpu_deficit={0: 100.0},
+            lanes=provider.lanes,
+            profiles=provider.profiles,
+            target_model_name="X",
+        )
+
+        # Control: self-eviction allowed when we don't identify the wakee.
+        control = planner._find_eviction_set(**common)
+        assert control, "setup must be tempting: lane is picked without the filter"
+        assert any(l.lane_id == "planner-X" for l, _a, _e in control)
```

Flake8 (7.3.0)

[error] ambiguous variable name 'l'

(E741)